### ISSUE TITLE
mget command

### ISSUE THREAD

**Issue by nuchiyama** [sewenew/redis-plus-plus#143]
I'd like to use "mget" command.

I wrote the following code.
```
#include <iostream>
#include <sw/redis++/redis++.h>

using namespace std;
using namespace sw::redis;

int main() {
 auto redis = Redis("tcp://127.0.0.1:6379");

 std::vector<std::string> keys;
 keys.push_back("key1");
 keys.push_back("key2");
 std::vector<OptionalString> res;
 redis.mget(keys.begin(), keys.end(), std::back_inserter(res));
 for (const auto &ele : res) {
  std::cout << *ele << std::endl;
 }
}
```
Compile Enviroment:
  CentOS7 / g++ (GCC) 8.3.1 20190311 (Red Hat 8.3.1-3)
Compile Option:
  g++ -std=c++17 -pthread -I/usr/local/redis-plus-plus/include -L/usr/local/redis-plus-plus/lib -lredis++ -L/usr/local/hiredis/lib -lhiredis a.cpp

But, I got error.
```
a.out: /usr/local/redis-plus-plus/include/sw/redis++/connection_pool.h:119: 
sw::redis::SafeConnection::SafeConnection(sw::redis::ConnectionPool&): Assertion `!_connection.broken()' failed.
Aborted
```

Please let me know how i should do.

Thank you.

**Comment by sewenew** [comment]
What's your hiredis and redis-plus-plus version? Did you build redis-plus-plus with `-DREDIS_PLUS_PLUS_CXX_STANDARD=17` option?

I cannot reproduce the problem. Also I cannot compile the code with the command you given. Instead, it works with the following command, e.g. put `-lredis++`, `-lhiredis` and `-pthread` after `a.cpp`:

```
g++ -std=c++17 -I/usr/local/redis-plus-plus/include -L/usr/local/redis-plus-plus/lib -L/usr/local/hiredis/lib a.cpp -lredis++ -lhiredis -pthread 
```

You can try this command to see if it works.

Regards

**Comment by nuchiyama** [comment]
I resolved the problem.

When I used "hiredis version 1.0.0" and "redis-plus-plus version 1.2.1", I got error.
And, I change "hiredis version 0.14.1", I go success.

Thank you so much !!

**Comment by brucejee** [comment]
I use gcc 4.8.5 and Redis-6.2.5, hiredis-1.0.2, redis-plus-plus-1.3.1, and get the same coredump, #61

**Comment by sewenew** [comment]
@brucejee Which build command did you use. From the above comments, it seems that the order of dynamic libraries might matter. Can you try the command I mentioned in the above comment to see if it works.

Regards

**Comment by brucejee** [comment]
@sewenew I fix the problem now, see this issue #61，thank you!
